README.md

```diff
@@ -1,84 +1,103 @@
+Below is a more complex WebSocket config example leveraging all of the above metadata options:
 
 ```js
-module.exports = {
+{
   websocket: {
     host: {
-      envVar: 'WS_HOST',
-      desc: 'Web Socket host url',
-      default: 'localhost'
+      description: 'WebSocket host URL',
+      default: 'localhost',
+      commandLineArg: '--ws-host',
+      envVar: 'WS_HOST'
     },
     port: {
-      envVar: 'WS_PORT',
-      desc: 'Web Socket port',
-      default: 8081
+      description: 'WebSocket port',
+      default: 8081,
+      commandLineArg: '--ws-port',
+      envVar: 'WS_PORT'
     },
     protocol: {
-      envVar: 'WS_PROTOCOL',
-      default: 'ws',
-      acceptedValues: {
-        values: ['ws', 'wss'],
-        fallback: 'ws'
-      },
-      desc: 'protocol for web sockets'
+      description: 'Which WebSocket protocol',
+      values: ['ws', 'wss'],
+      commandLineArg: '--ws-protocol',
+      envVar: 'WS_PROTOCOL'
     }
   }
 }
 ```
 
-A base object (primitive) in the object should have the three properties:
+### Use in your app
 
-* envVar: environment variable name
-* default: default value
-* desc: description
+Here's an example usage of source-configs using the schema defined above:
 
-As well, you can have the following optional properties:
+[TODO: This example is too complicated. We should not require users to do this much boilerplate.]
 
-* acceptedValues: a object to be used to define enumerations with a values field containing an array of accepted values and a fallback field which will be used if the config option is invalid
-* envVarParser: a function to parse environment variables or a string which will be used as a delimiter for simple delimiter split strings (This can only be implemented when writing the schema in a JS file or overwritten before being initialized)
+```js
+const sourceConfigs = require('source-configs')
 
-Also, a primitive can be a function which takes the parent scope config as a parameter to create strings based upon other config primitives.
+// Grab command line arguments. yargs-parser is used in this example but the minimum requirement is
+// a parser that converts command line arguments to a flat key-value map where the keys are camelCased
+const commandLineArgs = require('yargs-parser')(process.argv.slice(2))
 
-## Usage
+const config = sourceConfigs.init({ schema: require('./your-schema-js-file.js'), commandLineArguments: commandLineArgs })
 
-Here's an example usage of the module using the schema defined above
+// access one of the configs
+console.log(config.websocket.port)
+```
 
-```js
-const schema = require('./configSchema.js')
-const sourceConfigs = require('source-configs')
+[TODO: this should be all that is necessary. Note the removal of init.]
```

Comment:
The reasoning for the init function was that configs should be static across the runtime of the app, so rather than calling it multiple times across the application, it would be run once at startup and then you could access the cached config object anywhere without needing to source the configs again.

Second, on the situation of passing in command line arguments after they are parsed, I did such to give the user flexibility on what parser they wanted, but rather than saying you need to bring your own parser, baking in a command line args parser by default and exposing the option to add a custom one may be a better architecture choice.

Reply:
This was resolved by appending properties onto the sourceConfigs function itself